src/Workspaces/Core/Portable/Options/IOptionService.cs

```diff
@@ -61,6 +61,8 @@ internal interface IOptionService : IWorkspaceService
         /// </summary>
         IEnumerable<IOption> GetRegisteredOptions();
 
+        bool TryMapEditorConfigKeyToOption(string key, string? language, [NotNullWhen(true)] out IEditorConfigStorageLocation2? storageLocation, out OptionKey optionKey);
```

Comment:
Nit: Add doc comments for consistency with all methods in this interface.

Reply:
Actually, putting more thought and looking at https://github.com/dotnet/roslyn/pull/41872, I think it might be better to have this method on a separate interface, say `IEditorConfigToOptionMappingService` or some such. This way `IOptionService`, which has much broader functionality and also allows adding/removing/changing options and registering/unregistering workspace is not passed around within features.

Reply:
I left it for now. I'm not sure it saves much since you still need `IOptionService` for getting the options and it still needs to be a workspace service to avoid an `ExportProvider` requirement.